src/protean/core/entity.py

```diff
@@ -606,40 +666,68 @@ def create(cls, *args, **kwargs) -> 'Entity':
 
         model_cls, adapter = cls._retrieve_model()
 
-        # Build the entity from the input arguments
-        # Raises validation errors, if any, at this point
-        entity = cls(*args, **kwargs)
+        try:
```

Comment:
Why are we doing try-except in all these cases?

Reply:
Because I want to add helpful debug/log statements to lifecycle methods when they fail. The execution is a happy path today and does not handle errors that might happen in the adapter or database correctly.